src/Headers.ts

```diff
@@ -25,7 +25,7 @@ export class Headers {
      * because that class is only defined in the browser.
      */
     if (
-      ['Headers', 'HeadersPolyfill'].includes(init?.constructor.name) ||
+      ['Headers', 'HeadersPolyfill', '_Headers'].includes(init?.constructor.name) ||
```

Comment:
I like your proposal to check the `instanceof` against the global `Headers` class if it exists. Perhaps let's use it instead?

```js
|| (typeof globalThis.Headers !== 'undefined' && init instanceof globalThis.Headers)
```